File: bindings/python/example_replay.py
#!/usr/bin/env python3
"""
Replay example of using matgto-serge from Python

This example shows how to:
1. Create a proxy in replay mode
2. Replay a previously recorded cassette
3. Verify requests are served from cassette
"""

import sys
from matgto_serge import create_proxy, ProxyMode

def main():
    print("▶️  matgto-serge Python Example - Replay")
    print("=" * 50)

    # Create proxy instance in replay mode
    print("\n1️⃣ Creating proxy in REPLAY mode...")
    proxy = create_proxy("./cassettes")
    proxy = proxy.with_port(8888)
    proxy = proxy.with_mode(ProxyMode.REPLAY)
    print(f"   ✅ Proxy created on port {proxy.port()}")
    print(f"   ✅ Mode: {proxy.mode()}")

    # Replay cassette
    print("\n2️⃣ Loading cassette for replay...")
    cassette_name = "python-example"
    proxy.replay(cassette_name)
    print(f"   ✅ Replaying cassette: {cassette_name}")

    # Make HTTP requests (will be served from cassette)
    print("\n3️⃣ Making HTTP requests...")
    print("   Requests will be served from cassette (no real network calls)")
    print("   Configure your HTTP client:")
    print("   proxies={'http': 'http://localhost:8888', 'https': 'http://localhost:8888'}")

    # Example with requests library
    print("\n   Example code:")
    print("   ```python")
    print("   import requests")
    print("   response = requests.get(")
    print("       'https://httpbin.org/get',")
    print("       proxies={'http': 'http://localhost:8888'}")
    print("   )")
    print("   print(response.json())")
    print("   ```")

    # Shutdown
    print("\n4️⃣ Shutting down proxy...")
    proxy.shutdown()
    print("   ✅ Proxy stopped")

    print("\n" + "=" * 50)
    print("✅ Replay example complete!")

if __name__ == "__main__":
    try:
        main()
    except Exception as e:
        print(f"\n❌ Error: {e}", file=sys.stderr)
        sys.exit(1)

File: bindings/php/example_replay.php
<?php
/**
 * Replay example of using matgto-serge from PHP
 *
 * This example shows how to:
 * 1. Create a proxy in replay mode
 * 2. Replay a previously recorded cassette
 * 3. Verify requests are served from cassette
 */

require_once __DIR__ . '/MatgtoProxy.php';

use MatgtoSerge\MatgtoProxy;
use MatgtoSerge\ProxyMode;

echo "▶️  matgto-serge PHP Example - Replay\n";
echo str_repeat("=", 50) . "\n\n";

try {
    // Create proxy instance in replay mode
    echo "1️⃣ Creating proxy in REPLAY mode...\n";
    $proxy = new MatgtoProxy("./cassettes");
    $proxy->withPort(8888)
          ->withMode(ProxyMode::Replay);
    echo "   ✅ Proxy created on port " . $proxy->getPort() . "\n";
    echo "   ✅ Mode: " . $proxy->getMode()->value . "\n";

    // Replay cassette
    echo "\n2️⃣ Loading cassette for replay...\n";
    $cassetteName = "php-example";
    $proxy->replay($cassetteName);
    echo "   ✅ Replaying cassette: $cassetteName\n";

    // Make HTTP requests (will be served from cassette)
    echo "\n3️⃣ Making HTTP requests...\n";
    echo "   Requests will be served from cassette (no real network calls)\n";
    echo "   Configure your HTTP client:\n\n";

    echo "   Example with Guzzle:\n";
    echo "   ```php\n";
    echo "   use GuzzleHttp\\Client;\n";
    echo "   \n";
    echo "   \$client = new Client([\n";
    echo "       'proxy' => 'http://localhost:8888'\n";
    echo "   ]);\n";
    echo "   \n";
    echo "   \$response = \$client->get('https://httpbin.org/get');\n";
    echo "   echo \$response->getBody();\n";
    echo "   ```\n";

    echo "\n   Example with cURL:\n";
    echo "   ```php\n";
    echo "   \$ch = curl_init('https://httpbin.org/get');\n";
    echo "   curl_setopt(\$ch, CURLOPT_PROXY, 'http://localhost:8888');\n";
    echo "   curl_setopt(\$ch, CURLOPT_RETURNTRANSFER, true);\n";
    echo "   \$response = curl_exec(\$ch);\n";
    echo "   curl_close(\$ch);\n";
    echo "   echo \$response;\n";
    echo "   ```\n";

    // Shutdown
    echo "\n4️⃣ Shutting down proxy...\n";
    $proxy->shutdown();
    echo "   ✅ Proxy stopped\n";

    echo "\n" . str_repeat("=", 50) . "\n";
    echo "✅ Replay example complete!\n";

} catch (Exception $e) {
    echo "\n❌ Error: " . $e->getMessage() . "\n";
    exit(1);
}
